### PR TITLE
chore(rest): Making endpoints configurable

### PR DESCRIPTION
**Pull request by smrutis1**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.

Added a property in `application.yml` to put endpoints(endpoints URIs) and  http methods to be blacklisted in the format as shown below.

image

> * Which issue is this pull request belonging to and how is it solving it? (*#1924*)
> * Did you add or update any new dependencies that are required for your change?

Issue: 

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

**Comment by rudra-superrr** [comment]
Two things that I found while testing:
1. This won't work for id specific endpoints.
2. Taking spaces in URI and method when entering multiple endpoints, therefore trim the spaces.

**Comment by smrutis1** [comment]
> Two things that I found while testing:
> 
> 1. This won't work for id specific endpoints.
> 2. Taking spaces in URI and method when entering multiple endpoints, therefore trim the spaces.

@rudra-superrr , Kindly verify once.

**Comment by rudra-superrr** [comment]
Testing was successful.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/filter/EndpointsFilter.java [hunk 1]
```diff
@@ -19,25 +31,75 @@
...
-        } else {
+        String[] endpointMethodPairs = endpointsTobeBlackListed.split(",");
+        Map<String, Set<String>> endpointHttpMethods = getMapOfEndpointToHttpMethods(endpointMethodPairs);
+        boolean isAMatch = verifyMatchingOfRequestURIToEndpoints(requestURI, endpointHttpMethods);
+
+        if (!isAMatch) {
             filterChain.doFilter(request, response);
+        } else {
+            Set<String> httpMethodsToBeBlocked = new HashSet<>();
+            Optional<Entry<String, Set<String>>> vv = endpointHttpMethods.entrySet().stream().filter(es -> {
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/filter/EndpointsFilter.java:52]: Please use proper variable names for vv, ep.

Comment (smrutis1) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/filter/EndpointsFilter.java]: Done

**Comment by rudra-superrr** [comment]
Code looks good.

**Comment by rudra-superrr** [comment]
Following endpoints were tested:

image

**Pull request merged**